kythe/cxx/indexer/proto/testdata/proto_verifier_test.bzl

```diff
@@ -14,31 +14,53 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+load("@bazel_skylib//lib:collections.bzl", "collections")
 load(
     "@io_kythe//tools/build_rules/verifier_test:verifier_test.bzl",
     "KytheVerifierSources",
     "extract",
     "index_compilation",
     "verifier_test",
 )
+load("@rules_proto//proto:defs.bzl", "ProtoInfo")
 
 def _invoke(rulefn, name, **kwargs):
     """Invoke rulefn with name and kwargs, returning the label of the rule."""
     rulefn(name = name, **kwargs)
     return "//{}:{}".format(native.package_name(), name)
 
+def get_proto_files_and_proto_path_opts(protolibs):
+    """Given a list of proto_library targets, returns a list of all transitively
+    -included .proto files and a list of options to pass to protoc (containing
+    --proto_path locations).
+    """
+    srcs = []
+    proto_paths = []
+    for lib in protolibs:
+        if ProtoInfo not in lib:
+            fail("Expected a proto_library, got: %s" % str(lib))
+        for src in lib[ProtoInfo].transitive_sources.to_list():
+            if src.path.endswith(".proto"):
+                srcs.append(src)
+            proto_paths += lib[ProtoInfo].transitive_proto_path.to_list()
+    srcs = collections.uniq(srcs)
+    proto_path_opts = collections.before_each("--proto_path", collections.uniq(proto_paths))
```

Comment:
Curious: is there a possibility that identically named srcs in two different paths will now get confused? e.g. are we maintaining ordering on the proto_paths?

Reply:
I think that you're right that there is a possibility, although invoking `protoc` directly has the same limitations. I am maintaining the order from `ProtoInfo`, but because a `proto_library`s deps are not really ordered, I don't think it's well defined.

In the kythe repo, most `proto_library`s only have "--proto_path .", although any that depend on a `.proto` from an external dependency also have something like "bazel-out/k8-fastbuild/bin/external/com_google_protobuf/_virtual_imports/any_proto".